packages/svelte/test/preprocessors.test.ts

```diff
@@ -0,0 +1,174 @@
+import { componentTrackingPreprocessor, defaultComponentTrackingOptions } from '../src/preprocessors';
+
+function expectComponentCodeToBeModified(
+  preprocessedComponents: {
+    originalCode: string;
+    filename: string;
+    name: string;
+    newCode: string;
+    map: any;
+  }[],
+  options: any,
+) {
+  const expectedImportStmt = 'import { trackComponent } from "@sentry/svelte";\n';
+
+  preprocessedComponents.forEach(cmp => {
+    const expectedFunctionCallOptions = {
+      trackInit: options?.trackInit ?? true,
+      trackUpdates: options?.trackUpdates ?? true,
+      componentName: cmp.name,
+    };
+    const expectedFunctionCall = `trackComponent(${JSON.stringify(expectedFunctionCallOptions)});\n`;
+    expect(cmp.newCode).toBeDefined();
+    expect(cmp.map).toBeDefined();
+    expect(cmp.newCode).toEqual(`${expectedImportStmt}${expectedFunctionCall}${cmp.originalCode}`);
+  });
+}
+
+describe('componentTrackingPreprocessor', () => {
```

Comment:
Very nice tests! Would it make sense to do one more tests where we check that the code is not injected multiple times if there are multiple script tags?

Reply:
Yup, I added this test + another one just before requesting the review. https://github.com/getsentry/sentry-javascript/blob/e2e081d866cccb494ef5c02064ed3ace55a1686f/packages/svelte/test/preprocessors.test.ts#L116

I can only test this by calling the preprocessor twice on the same component because that's what would happen in a real-world scenario as well. The `script` preprocessor only gets the contet of one `<script>` tag at a time.

Reply:
Ah got it. I thought we would have to pass in a string like `<script>...</script><script>...</script>`. Thanks for explaining!